core/templates/services/user-backend-api.service.ts

```diff
@@ -136,19 +132,54 @@ export class UserBackendApiService {
     }).toPromise();
   }
 
-  async getPreferencesAsync(): Promise<PreferencesBackendDict> {
-    return this.http.get<PreferencesBackendDict>(this.PREFERENCES_DATA_URL)
-      .toPromise();
+  async getPreferencesAsync():
+    Promise<NonEmailPreferencesBackendDict> {
+    return this.http.get<NonEmailPreferencesBackendDict>(
+      this.PREFERENCES_DATA_URL).toPromise();
+  }
+
+  async getPreferencesProfilePictureDataUrlAsync():
+    Promise<ProfilePictureDataBackendDict> {
+    return this.http.get<ProfilePictureDataBackendDict>(
+      this.PREFERENCES_PROFILE_PICTURE_DATA_URL).toPromise();
+  }
+
+  async getEmailPreferencesAsync():
+    Promise<EmailPreferencesBackendDict> {
+    return this.http.get<EmailPreferencesBackendDict>(
+      this.EMAIL_PREFFERENCES).toPromise();
   }
 
   async updatePreferencesDataAsync(
       updateType: string,
-      data: boolean | string | string[] | EmailPreferencesBackendDict
+      data: boolean | string | string[]
+  ): Promise<NonEmailPreferencesBackendDict> {
+    return this.http.put<NonEmailPreferencesBackendDict>(
+      this.PREFERENCES_DATA_URL, {
+        update_type: updateType,
+        data: data
+      }
+    ).toPromise();
+  }
+
+  async updatePreferencesProfilePictureDataUrlAsync(
+      data: string
+  ): Promise<ProfilePictureDataBackendDict> {
+    return this.http.put<ProfilePictureDataBackendDict>(
+      this.PREFERENCES_PROFILE_PICTURE_DATA_URL, {
+        data: data
+      }
+    ).toPromise();
+  }
+
+  async updateEmailPreferencesAsync(
+      data: boolean | EmailPreferencesBackendDict
```

Comment:
Shouldn't this be only `EmailPreferencesBackendDict`

Reply:
yes, updated